stackslib/src/chainstate/stacks/boot/signers-voting.clar

```diff
@@ -0,0 +1,102 @@
+;;
+;; @contract voting for the aggregate public key
+;;
+
+;; maps dkg round and signer to proposed aggregate public key
+(define-map votes {reward-cycle: uint, round: uint, signer: principal} {aggregate-public-key: (buff 33), reward-slots: uint})
+;; maps dkg round and aggregate public key to weights of signers supporting this key so far
+(define-map tally {reward-cycle: uint, round: uint, aggregate-public-key: (buff 33)} uint)
+;; maps aggregate public keys to rewards cycles and rounds
+(define-map used-aggregate-public-keys (buff 33) {reward-cycle: uint, round: uint})
+
+(define-constant err-signer-index-mismatch (err u10000))
+(define-constant err-invalid-signer-index (err u10001))
+(define-constant err-out-of-voting-window (err u10002))
+(define-constant err-old-round (err u10003))
+(define-constant err-ill-formed-aggregate-public-key (err u10004))
+(define-constant err-duplicate-aggregate-public-key (err u10005))
+(define-constant err-duplicate-vote (err u10006))
+(define-constant err-invalid-burn-block-height (err u10007))
+
+(define-constant pox-info
+    (unwrap-panic (contract-call? .pox-4 get-pox-info)))
+
+;; maps reward-cycle ids to last round
+(define-map rounds uint uint)
+
+(define-data-var state-1 {reward-cycle: uint, round: uint, aggregate-public-key: (optional (buff 33)),
+    total-votes: uint}  {reward-cycle: u0, round: u0, aggregate-public-key: none, total-votes: u0})
+(define-data-var state-2 {reward-cycle: uint, round: uint, aggregate-public-key: (optional (buff 33)),
+    total-votes: uint}  {reward-cycle: u0, round: u0, aggregate-public-key: none, total-votes: u0})
+
+;; get voting info by burn block height
+(define-read-only (get-info (height uint))
+    (ok (at-block (unwrap! (get-block-info? id-header-hash height) err-invalid-burn-block-height) (get-current-info))))
+
+;; get current voting info
+(define-read-only (get-current-info)
+    (if (is-eq u0 (mod (current-reward-cycle) u2)) (var-get state-1) (var-get state-2)))
```

Comment:
Per discussion with @kantai. Since the slot switching is currently not worked out he said to only implement against the current state of affairs; i.e., a single list of signers.

Issue: #4307

Reply:
Info about the state is not used anywhere.

Reply:
now only state-1 is used